python/kotekan/visbuffer.py

```diff
@@ -299,38 +300,236 @@ class VisRaw(object):
+        num_time
+            Number of time samples in the buffer.
+        num_freq
+            Number of frequencies in the buffer.
+
+        Returns
+        -------
+        VisRaw
+            VisRaw viewing the data in the buffer.
+
+        Raises
+        ------
+        ValueError
+            If there was a problem parsing the buffer into the VisRaw structure.
+        """
+        # Create a simple struct to access the metadata (num_elements, num_prod, num_ev)
+        align_valid = 4  # valid field is 4 byte aligned
+        frame_struct_simple = np.dtype(
+            {
+                "names": ["valid", "metadata", "data"],
+                "formats": [
+                    np.uint8,
+                    VisMetadata,
+                    (np.void, size_frame - align_valid - ctypes.sizeof(VisMetadata)),
+                ],
+                "offsets": [0, align_valid, align_valid + ctypes.sizeof(VisMetadata)],
+                "itemsize": size_frame,
+            }
+        )
+        raw = buffer.view(dtype=frame_struct_simple)
+        num_elements = np.unique(
+            raw["metadata"][raw["valid"].astype(np.bool)]["num_elements"]
+        )
+        if len(num_elements) > 1:
+            raise ValueError(
+                "Found more than 1 value for `num_elements` in numpy ndarray: {}.".format(
+                    num_elements
+                )
+            )
+        num_elements = num_elements[0]
+        num_prod = np.unique(raw["metadata"][raw["valid"].astype(np.bool)]["num_prod"])
+        if len(num_prod) > 1:
+            raise ValueError(
+                "Found more than 1 value for `num_prod` in numpy ndarray: {}.".format(
+                    num_prod
+                )
+            )
+        num_prod = num_prod[0]
+        num_ev = np.unique(raw["metadata"][raw["valid"].astype(np.bool)]["num_ev"])
+        if len(num_ev) > 1:
+            raise ValueError(
+                "Found more than 1 value for `num_ev` in numpy ndarray: {}.".format(
+                    num_ev
+                )
+            )
+        num_ev = num_ev[0]
+
+        # Now that we have some metadata, we can really parse the data...
+        frame_struct = cls.frame_struct(
+            size_frame, num_elements, num_prod, num_ev, align_valid=True
+        )
+
+        # TODO: Python 3 - use native_str for compatibility
+        raw = buffer.view(dtype=frame_struct)
+        data = raw["data"]
+        metadata = raw["metadata"]
+        valid_frames = raw["valid"]
+
+        ctime = metadata["ctime"]
+        fpga_seq = metadata["fpga_seq"]
+        num_prod = metadata["num_prod"]
+
+        time = np.ndarray(
+            shape=(num_time, num_freq),
+            dtype=[
+                (native_str("fpga_count"), np.uint64),
+                (native_str("ctime"), time_spec),
+            ],
+        )
+        for t in range(num_time):
+            for f in range(num_freq):
+                time[t, f] = (fpga_seq[t, f], ctime[t, f])
```

Comment:
This doesn't work anymore, because `metadata["ctime"]` has to be converted from `time_spec` to float. I don't think there's a way to still do it like this.